deploy/heroku-deploy/Dockerfile

```diff
@@ -1,8 +1,8 @@
-FROM heroku/heroku:18
+FROM ruby:3.1.3
```

Comment:
`heroku-18` comes with Ruby 2.5, so that's what we've been using this whole time. We don't need any of the extra stuff that comes in the Heroku base images, so I just switched to a Ruby image.